st2api/st2api/controllers/resource.py

```diff
@@ -129,7 +132,14 @@ def _get_all(self, exclude_fields=None, **kwargs):
             pecan.response.headers['X-Limit'] = str(limit)
         pecan.response.headers['X-Total-Count'] = str(instances.count())
 
-        return [self.model.from_model(instance) for instance in instances[offset:eop]]
+        from_model_kwargs = self._get_from_model_kwargs_for_request(request=pecan.request)
+
+        result = []
+        for instance in instances[offset:eop]:
+            item = self.model.from_model(instance, **from_model_kwargs)
```

Comment:
Yeah, it's really just a question of semantics - which layer should throw, which one should consume exception, which one should format them for the response...